### PR TITLE
Add documentation about Action Cable npm package

### PR DESCRIPTION
**Pull request by maclover7**
Sorry, forgot to include in my main PR :(

[ci skip]

_Jon Moss, Zach Schneider_

cc @jeremy @javan

### CODE REVIEW

File: actioncable/README.md [hunk 1]
```diff
@@ -460,6 +460,28 @@ with all the popular application servers -- Unicorn, Puma and Passenger.
 Action Cable does not work with WEBrick, because WEBrick does not support the
 Rack socket hijacking API.
 
+## npm usage
+
+In addition to being available through the `actioncable` gem, Action Cable's frontend JS assets are also bundled in an officially supported npm
```
Comment (rafaelfranca) [inline actioncable/README.md]: This line is too long

File: actioncable/README.md [hunk 2]
```diff
@@ -460,6 +460,28 @@ with all the popular application servers -- Unicorn, Puma and Passenger.
...
+module, intended for usage in standalone frontends that communicate with a Rails
+API.
```
Comment (jeremy) [inline actioncable/README.md]: Go broader. Discuss asset packaging options. Most apps will use Action Cable with the asset pipeline (…and here's how) but you can use other asset packagers, too (and here's how).

Comment (vipulnsward) [inline actioncable/README.md]: Good relative documentation available from turbolinks docs.

**Comment by maclover7** [comment]
sorry for the delay @jeremy, updated 👍

File: actioncable/README.md [hunk 3]
```diff
@@ -460,6 +460,74 @@ with all the popular application servers -- Unicorn, Puma and Passenger.
...
+
+```coffeescript
+#= require action_cable
+```
+
+### npm usage
+
+In addition to being available through the `actioncable` gem, Action Cable's
+frontend JS assets are also bundled in an officially supported npm module,
+intended for usage in standalone frontends that communicate with a Rails application.
```
Comment (vipulnsward) [inline actioncable/README.md]: frontend Applications

File: actioncable/README.md [hunk 4]
```diff
@@ -460,6 +460,74 @@ with all the popular application servers -- Unicorn, Puma and Passenger.
...
+A common use case for this could be if you have a decoupled frontend application
+written in React, Ember, etc. and want to add real-time WebSocket functionality,
```
Comment (vipulnsward) [inline actioncable/README.md]: Ember.js

File: actioncable/README.md [hunk 5]
```diff
@@ -460,6 +460,74 @@ with all the popular application servers -- Unicorn, Puma and Passenger.
...
+that talks with your Rails API.
```
Comment (vipulnsward) [inline actioncable/README.md]: This is a bit out of context, lets drop this end line.

**Comment by maclover7** [comment]
updated 👍

File: actioncable/README.md [hunk 6]
```diff
@@ -460,6 +460,74 @@ with all the popular application servers -- Unicorn, Puma and Passenger.
...
+The `ActionCable` constant is available as a `require`-able module, so
+you only have to require the package to gain access to the API that is
+provided.
+
+In JavaScript...
+
+```javascript
+ActionCable = require('actioncable')
+
+cable = ActionCable.createConsumer('wss://RAILS-API-PATH.com/cable')
```
Comment (vipulnsward) [inline actioncable/README.md]: var cable

**Comment by maclover7** [comment]
👍 we should push for global variables rather than instantiating consumers all over the place ;)

**Comment by vipulnsward** [comment]
@maclover7 ideally one will instantiate and export them. There is no "global" per say in npm App, and its relative.

**Comment by maclover7** [comment]
👍 yep, this is just meant to get people started

**Comment by vipulnsward** [comment]
Getting this in for the release, since related changes are already in as part of https://github.com/rails/rails/pull/24991. 
cc @rafaelfranca

**Pull request merged**